### DISCUSSION TITLE
invokeDirectiveHook Cannot read properties of undefined (reading 'value')

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#16851] [Questions] [answered]
Hey there.
After upgrading from rc8 to rc9 i got this error:

```
runtime-core.esm-bundler.js:2779 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'value')
    at invokeDirectiveHook (runtime-core.esm-bundler.js:2779:47)
    at Array.<anonymous> (runtime-core.esm-bundler.js:5349:25)
    at flushPostFlushCbs (runtime-core.esm-bundler.js:339:32)
    at flushJobs (runtime-core.esm-bundler.js:393:9)
```

This error triggering after router navigation.

![image](https://d3dehtdmp2rwcw.cloudfront.net/ms_450033/Q1qizeJxRYlCZis50pQMOKXjBZLgpg/Monosnap%2Bonway.localreg%2B2022-09-06%2B20-05-59.png?Expires=1662485400&Signature=JltLHIMHXTp4W~YdjLDfU7CSmXFETDjMKh4KZfksd5B7-tybnvoEDNp5plPvIvFXH9tdrlZbJCWxmONS3~c399CaptD3flMvbhOeuluFgl0s9XR8uO5BwPnytnzWjurq2nS-qcBJNz06iLBVmrIjsbJWay~GNVoAh98csCqFXAFGjMzreebweJRKZ7-iu1WVcvTbLJNbbp3TlMyNUQCc8ZItnvsrKavf9sfFthdtRm2jLqe0C~dL8wIr3aMD9E2jcId6WCMH8L-27Ot5VX6GJGXB6HDNE5Rzf2QBa7piQXLXZD3cWCl7eafrkf5HePn3vUC0O6OhbvjDUL7iwzXXNA__&Key-Pair-Id=APKAJBCGYQYURKHBGCOA)

How i can fix it ? 
Thanks

**Comment by danielroe** [comment]
This seems like an issue with a directive but it's not possible to say much more without seeing some code.

**Reply by jhughes-dev** [reply]
I am seeing a similar issue. It happens sporadically. I don't see any route navigation causing this. My app will work, and then it hits this error and halts.

I am heavily using Vuetify3 beta as well. 

I can try to work out an example without that but I am suspicious of this code:
```
function invokeDirectiveHook(vnode, prevVNode, instance, name) {
    const bindings = vnode.dirs;
    const oldBindings = prevVNode && prevVNode.dirs;
    for (let i = 0; i < bindings.length; i++) {
        const binding = bindings[i];
        if (oldBindings) {
            binding.oldValue = oldBindings[i].value;
        }
        let hook = binding.dir[name];
        if (hook) {
            // disable tracking inside all lifecycle hooks
            // since they can potentially be called inside effects.
            pauseTracking();
            callWithAsyncErrorHandling(hook, instance, 8 /* ErrorCodes.DIRECTIVE_HOOK */, [
                vnode.el,
                binding,
                vnode,
                prevVNode
            ]);
            resetTracking();
        }
    }
}
```

specifically this check:

``` 
        if (oldBindings) {
            binding.oldValue = oldBindings[i].value;
        }
```
It's checking `oldBindings`, but assumes it's an array of `{value: unknown}` at the least. 

It looks like `prevVNode.dirs` may not have a value property or may be something other than an array. I assume `dirs` is directives. 

When I try this with a breakpoint in place, I haven't seen this issue, and I can't get it to pause on exceptions either.

**Reply by jhughes-dev** [reply]
Again, completely naive thought. Is prevVNode.dirs the same length as vnode.dirs?

**Reply by jhughes-dev** [reply]
I caught this case with a conditional breakpoint after the naïve thought. The two arrays are not the same length in all cases. 
![Untitled-1](https://user-images.githubusercontent.com/23299787/189436434-dd1981ab-9048-40ba-9099-f2bb0ea03b6d.png)

**Reply by jhughes-dev** [reply]
I don't even know if that's nuxt code, or Vue code.

**Reply by k0mar12** [reply]
@jhughes-dev Thanks for ur reply. This is vue component binding code. Did u use v-img component? Coz in my case this component breaking.

**Reply by k0mar12** [reply]
@danielroe Sorry, problem, probably, in vuetify. Thanks :)

**Reply by jhughes-dev** [reply]
I did try taking out the v-img in favor of a standard img tag and that stopped the error from appearing, but I still think this is a core vue issue. That code isn't robust.

**Reply by k0mar12** [reply]
@jhughes-dev They fixed it in this [commit](https://github.com/vuetifyjs/vuetify/commit/3c071db095599f946be29278081bd3c9ae156daa)

**Comment by jhughes-dev** [comment, marked as answer]
I think it's this vue core issue.  
You could try winding back the vue version a couple of points.
https://github.com/vuejs/core/issues/6573

**Reply by k0mar12** [reply]
@jhughes-dev fixed in vuetify rc 11
